synapse/tools/axon/dump.py

```diff
@@ -0,0 +1,180 @@
+import os
+import sys
+import asyncio
+import hashlib
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+import synapse.telepath as s_telepath
+
+import synapse.lib.cmd as s_cmd
+import synapse.lib.coro as s_coro
+import synapse.lib.const as s_const
+import synapse.lib.output as s_output
+import synapse.lib.msgpack as s_msgpack
+
+descr = '''
+Dump blobs from a Synapse Axon.
+'''
+
+DEFAULT_ROTATE_SIZE = s_const.gigabyte * 4
+
+def getOutfileName(celliden, start, end):
+    return f'{celliden}.{start:012d}-{end:012d}.blobs'
+
+def writeFooterAndClose(outfile, celliden, file_start, file_end, file_blobcount, outfile_path, outdir, outp):
+    footer = s_msgpack.en(('blob:fini', {'celliden': celliden,
+                                         'start': file_start,
+                                         'end': file_end,
+                                         'count': file_blobcount,
+                                         'tock': s_common.now()}))
+    outfile.write(footer)
+    final_name = os.path.join(outdir, getOutfileName(celliden, file_start, file_end))
+    outfile.close()
+    os.rename(outfile_path, final_name)
+    outp.printf(f'Wrote {file_blobcount} blobs to {final_name}')
+
+async def dumpBlobs(opts, outp):
+
+    try:
+        async with await s_telepath.openurl(opts.url) as axon:
+
+            cellinfo = await axon.getCellInfo()
+            celltype = cellinfo['cell']['type']
+            if "axon" not in celltype.lower():
+                mesg = f'Axon dump tool only works on axons, not {celltype}'
+                raise s_exc.TypeMismatch(mesg=mesg)
+
+            celliden = cellinfo['cell']['iden']
+            cellvers = cellinfo['cell']['version']
+
+            if os.path.exists(opts.outdir) and not os.path.isdir(opts.outdir):
+                mesg = f'Specified output directory {opts.outdir} exists but is not a directory.'
+                raise s_exc.BadDataValu(mesg=mesg)
+            os.makedirs(opts.outdir, exist_ok=True)
+
+            start = opts.offset
+            rotate_size = opts.rotate_size
+            hashes_iter = axon.hashes(start)
+            last_offset = start
+            blobcount = 0
+            outfile = None
+            outfile_path = None
+            file_start = start
+            file_size = 0
+            for_open = True
+
+            state = {}
+            if opts.statefile is not None:
+                statefile_path = opts.statefile
+                if os.path.isdir(statefile_path):
+                    statefile_path = os.path.join(statefile_path, f'{celliden}.yaml')
+                if (data := s_common.yamlload(statefile_path)) is not None:
+                    state = data
+                opts.statefile = statefile_path
```

Comment:
This only creates the state file if this option was specified on the command line. The layer dump tool always create the state file and I think this should also.

Recommend using similar behavior to the layer dump tool where it stores the state file in the outdir by default.